File: src/main/java/com/raoulvdberge/refinedstorage/apiimpl/network/node/CableNetworkNode.java
package com.raoulvdberge.refinedstorage.apiimpl.network.node;

import com.raoulvdberge.refinedstorage.RS;
import net.minecraft.util.ResourceLocation;
import net.minecraft.util.math.BlockPos;
import net.minecraft.world.World;

public class CableNetworkNode extends NetworkNode {
    public static final ResourceLocation ID = new ResourceLocation(RS.ID, "cable");

    public CableNetworkNode(World world, BlockPos pos) {
        super(world, pos);
    }

    @Override
    public int getEnergyUsage() {
        return RS.SERVER_CONFIG.getCable().getUsage();
    }

    @Override
    public ResourceLocation getId() {
        return ID;
    }
}

File: src/main/kotlin/com/refinedmods/refinedstorage/apiimpl/network/node/CableNetworkNode.kt
package com.refinedmods.refinedstorage.apiimpl.network.node

import com.refinedmods.refinedstorage.RS
import net.minecraft.util.Identifier
import net.minecraft.util.math.BlockPos
import net.minecraft.world.World

class CableNetworkNode(
        world: World,
        pos: BlockPos,
        override val id: Identifier = ID
): NetworkNode(world, pos) {
    companion object {
        val ID: Identifier = Identifier(RS.ID, "cable")
    }

    override val energyUsage: Int
        get() = RS.SERVER_CONFIG.cable.getUsage()

}